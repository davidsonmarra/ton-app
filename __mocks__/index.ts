export { productDetailsApi } from './product-details-api';
export { productsApi } from './products-api';
