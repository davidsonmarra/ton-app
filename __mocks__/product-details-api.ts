export const productDetailsApi = [
  {
    id: 1,
    title: 'T1',
    subTitle: 'Promo',
    oldPrice: '198,80',
    currentPrice: '106,80',
    installmentsValue: '8',
    installmentsValueCents: '90',
    images: [
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t1-1',
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t1-2',
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t1-3'
    ],
    benefits: [
      {
        icon: 'calendar-blank-outline',
        description: 'Receba suas vendas em 1 dia útil'
      },
      {
        icon: 'truck-outline',
        description: 'Frete grátis para todo o Brasil'
      },
      {
        icon: 'cellphone-check',
        description: 'Precisa de celular com bluetooth e conectado'
      },

      {
        icon: 'chat-alert-outline',
        description: 'Comprovante via aplicativo e SMS'
      }
    ]
  },
  {
    id: 2,
    title: 'T1 Chip',
    subTitle: 'Promo',
    oldPrice: '318,80',
    currentPrice: '286,80',
    installmentsValue: '23',
    installmentsValueCents: '90',
    images: [
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t1-chip-1',
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t1-chip-2',
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t1-chip-3'
    ],
    benefits: [
      {
        icon: 'calendar-blank-outline',
        description: 'Receba suas vendas em 1 dia útil'
      },
      {
        icon: 'truck-outline',
        description: 'Frete grátis para todo o Brasil'
      },
      {
        icon: 'cellphone-remove',
        description: 'Não precisa de um celular'
      },
      {
        icon: 'chat-alert-outline',
        description: 'Comprovante via aplicativo e SMS'
      },
      {
        icon: 'wifi',
        description: 'Conexão Wi-Fi e chip de dados'
      },
      {
        icon: 'nfc',
        description: 'Pagamento por aproximação (NFC)'
      }
    ]
  },
  {
    id: 3,
    title: 'T2+',
    subTitle: 'Promo',
    oldPrice: '478,80',
    currentPrice: '190,80',
    installmentsValue: '15',
    installmentsValueCents: '90',
    images: [
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t2-1',
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t2-2',
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t2-3'
    ],
    benefits: [
      {
        icon: 'calendar-blank-outline',
        description: 'Receba suas vendas em 1 dia útil'
      },
      {
        icon: 'truck-outline',
        description: 'Frete grátis para todo o Brasil'
      },
      {
        icon: 'cellphone-remove',
        description: 'Não precisa de um celular'
      },
      {
        icon: 'chat-alert-outline',
        description: 'Comprovante via aplicativo e SMS'
      },
      {
        icon: 'wifi',
        description: 'Conexão Wi-Fi e chip de dados'
      },
      {
        icon: 'nfc',
        description: 'Pagamento por aproximação (NFC)'
      },
      {
        icon: 'battery-high',
        description: 'Bateria de longa duração (+ de 12h)'
      }
    ]
  },
  {
    id: 4,
    title: 'T3',
    subTitle: 'Promo',
    oldPrice: '958,80',
    currentPrice: '394,80',
    installmentsValue: '32',
    installmentsValueCents: '90',
    images: [
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t3-1',
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t3-2',
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t3-3'
    ],
    benefits: [
      {
        icon: 'calendar-blank-outline',
        description: 'Receba suas vendas em 1 dia útil'
      },
      {
        icon: 'truck-outline',
        description: 'Frete grátis para todo o Brasil'
      },
      {
        icon: 'cellphone-remove',
        description: 'Não precisa de um celular'
      },
      {
        icon: 'newspaper',
        description: 'Comprovante impresso ou SMS'
      },
      {
        icon: 'wifi',
        description: 'Conexão Wi-Fi e chip de dados'
      },
      {
        icon: 'nfc',
        description: 'Pagamento por aproximação (NFC)'
      },
      {
        icon: 'qrcode',
        description: 'Aceita Pix QR Code na Maquininha'
      }
    ]
  },
  {
    id: 5,
    title: 'T3 Smart',
    subTitle: 'Promo',
    oldPrice: '958,80',
    currentPrice: '478,80',
    installmentsValue: '39',
    installmentsValueCents: '90',
    images: [
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_100/site-ton/maquininhas/release-machine-t3-smart-1',
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_100/site-ton/maquininhas/release-machine-t3-smart-2',
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_100/site-ton/maquininhas/release-machine-t3-smart-3'
    ],
    benefits: [
      {
        icon: 'calendar-blank-outline',
        description: 'Receba suas vendas em 1 dia útil'
      },
      {
        icon: 'truck-outline',
        description: 'Frete grátis para todo o Brasil'
      },
      {
        icon: 'qrcode',
        description: 'Aceita Pix QR Code na Maquininha'
      },
      {
        icon: 'cellphone-remove',
        description: 'Não precisa de um celular'
      },
      {
        icon: 'newspaper',
        description: 'Comprovante impresso ou SMS'
      },
      {
        icon: 'nfc',
        description: 'Pagamento por aproximação (NFC)'
      },
      {
        icon: 'wifi',
        description: 'Conexão Wi-Fi e chip de dados'
      },
      {
        icon: 'battery-high',
        description: 'Bateria de longa duração'
      },
      {
        icon: 'android',
        description: 'Sistema Android com Visor Touchscreen'
      }
    ]
  }
];
