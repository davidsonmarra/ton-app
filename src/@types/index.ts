export type { ProductDTO } from './productDTO';
export type { ProductsStackParamList } from './routes';
