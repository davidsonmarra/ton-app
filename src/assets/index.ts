export { default as T1Img } from './t1.png';
