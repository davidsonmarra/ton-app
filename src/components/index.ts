export { Button } from './button';
export { BottomModal } from './bottom-modal';
export { BenefitItemDetails } from './benefit-item-details';
export { Header } from './header';
export { ImageCardSelect } from './image-card-select';
export { ImageDetails } from './image-details';
export { PriceDetails } from './price-details';
export { ProductCard } from './product-card';
export { TitleProductDetails } from './title-product-details';
