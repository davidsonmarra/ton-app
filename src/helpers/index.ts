export { productDetailsApi } from './productDetailsApi';
export { productsApi } from './productsApi';
