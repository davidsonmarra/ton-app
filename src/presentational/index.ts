export { ProductDetails } from './product-details';
export { ProductsList } from './products-list';
