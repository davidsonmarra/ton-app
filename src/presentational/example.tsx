import { Header } from '../components';

export const Example = () => {
  return <Header title='PÃ¡gina de Exemplo' />;
};
