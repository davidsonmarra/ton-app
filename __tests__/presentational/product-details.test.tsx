import { productDetailsApi } from '../../__mocks__';
import { ProductDetails } from '../../src/presentational';
import { matchSnapshotWithProviders, render } from '../test-utils';

describe('ProductDetails', () => {
  it('should match a snapshot', () => {
    matchSnapshotWithProviders(
      <ProductDetails
        isLoadingApi={false}
        item={productDetailsApi[0]}
        handlePressLeftIcon={jest.fn}
        bottomInset={0}
      />
    );
  });

  it('should render a loading indicator', () => {
    const { getByTestId } = render(
      <ProductDetails
        isLoadingApi={true}
        item={productDetailsApi[0]}
        handlePressLeftIcon={jest.fn}
        bottomInset={0}
      />
    );
    expect(getByTestId('loading-indicator')).toBeTruthy();
  });

  it('should render a product details', () => {
    const { getByText } = render(
      <ProductDetails
        isLoadingApi={false}
        item={productDetailsApi[0]}
        handlePressLeftIcon={jest.fn}
        bottomInset={0}
      />
    );
    expect(getByText(productDetailsApi[0].title)).toBeTruthy();
    expect(getByText(productDetailsApi[0].subTitle)).toBeTruthy();
  });
});
