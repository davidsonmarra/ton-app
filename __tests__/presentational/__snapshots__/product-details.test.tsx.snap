// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetails should match a snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "off",
    }
  }
  style={
    {
      "backgroundColor": "#efefef",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#1Ec677",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
      }
    }
    topInset={0}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": 32,
        }
      }
      testID="header-left-button"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#fff",
              "fontSize": 24,
            },
            undefined,
            {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
        testID="header-left-icon"
      >
        
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#fff",
          "fontFamily": "Poppins-Bold",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      T1  Promo
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": 32,
        }
      }
      testID="header-button-right"
    />
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingBottom": 32,
        "paddingHorizontal": 24,
        "paddingTop": 24,
      }
    }
    showsVerticalScrollIndicator={false}
    style={{}}
  >
    <View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            {
              "color": "#1e1e1e",
              "fontFamily": "Poppins-Bold",
              "fontSize": 24,
            }
          }
        >
          T1
        </Text>
        <Text
          style={
            {
              "color": "#1e1e1e",
              "fontFamily": "Poppins-Medium",
              "fontSize": 24,
            }
          }
        >
           
          Promo
        </Text>
      </View>
      <View
        style={
          {
            "marginTop": 24,
          }
        }
      />
      <Image
        resizeMode="contain"
        source={
          {
            "uri": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t1-1",
          }
        }
        style={
          {
            "height": 250,
            "marginBottom": 24,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 0,
            "width": 250,
          }
        }
        testID="image"
      />
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 0,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 0,
            "width": 250,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#fff",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderColor": "#1Ec677",
              "borderStyle": "solid",
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "borderWidth": 2,
              "elevation": 9,
              "height": 100,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.32,
              "shadowRadius": 5.46,
              "width": 71.42857142857143,
            }
          }
          testID="image-card-select-button"
        >
          <Image
            resizeMode="contain"
            source={
              {
                "uri": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t1-1",
              }
            }
            style={
              {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              }
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#fff",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderColor": "transparent",
              "borderStyle": "solid",
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "borderWidth": 2,
              "elevation": 9,
              "height": 100,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.32,
              "shadowRadius": 5.46,
              "width": 71.42857142857143,
            }
          }
          testID="image-card-select-button"
        >
          <Image
            resizeMode="contain"
            source={
              {
                "uri": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t1-2",
              }
            }
            style={
              {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              }
            }
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#fff",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderColor": "transparent",
              "borderStyle": "solid",
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "borderWidth": 2,
              "elevation": 9,
              "height": 100,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.32,
              "shadowRadius": 5.46,
              "width": 71.42857142857143,
            }
          }
          testID="image-card-select-button"
        >
          <Image
            resizeMode="contain"
            source={
              {
                "uri": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_384,q_auto/site-ton/maquininhas/machine-t1-3",
              }
            }
            style={
              {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          {
            "marginTop": 24,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 8,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#1Ec677",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            󰭦
          </Text>
          <Text
            style={
              {
                "color": "#1e1e1e",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Poppins-Regular",
                "fontSize": 16,
                "marginLeft": 8,
              }
            }
          >
            Receba suas vendas em 1 dia útil
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 8,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#1Ec677",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            󱊝
          </Text>
          <Text
            style={
              {
                "color": "#1e1e1e",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Poppins-Regular",
                "fontSize": 16,
                "marginLeft": 8,
              }
            }
          >
            Frete grátis para todo o Brasil
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 8,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#1Ec677",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            󱟽
          </Text>
          <Text
            style={
              {
                "color": "#1e1e1e",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Poppins-Regular",
                "fontSize": 16,
                "marginLeft": 8,
              }
            }
          >
            Precisa de celular com bluetooth e conectado
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 8,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#1Ec677",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            󱋉
          </Text>
          <Text
            style={
              {
                "color": "#1e1e1e",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Poppins-Regular",
                "fontSize": 16,
                "marginLeft": 8,
              }
            }
          >
            Comprovante via aplicativo e SMS
          </Text>
        </View>
        <View
          style={
            {
              "marginBottom": 24,
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 24,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "marginRight": 24,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#656565",
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 12,
                    "textDecorationColor": "black",
                    "textDecorationLine": "line-through",
                    "textDecorationStyle": "solid",
                  }
                }
              >
                R$ 
                198,80
              </Text>
              <Text
                style={
                  {
                    "color": "#1e1e1e",
                    "fontFamily": "Poppins-SemiBold",
                    "fontSize": 16,
                  }
                }
              >
                R$ 
                106,80
              </Text>
              <Text
                style={
                  {
                    "color": "#656565",
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 12,
                  }
                }
              >
                à vista ou
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={{}}
              >
                <Text
                  style={
                    {
                      "color": "#1e1e1e",
                      "fontFamily": "Poppins-SemiBold",
                      "fontSize": 12,
                    }
                  }
                >
                  12x
                </Text>
                <Text
                  style={
                    {
                      "color": "#1Ec677",
                      "fontFamily": "Poppins-SemiBold",
                      "fontSize": 12,
                    }
                  }
                >
                  R$
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "flex-start",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#1Ec677",
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 64,
                    }
                  }
                >
                  8
                </Text>
                <Text
                  style={
                    {
                      "color": "#1Ec677",
                      "fontFamily": "Poppins-SemiBold",
                      "fontSize": 16,
                    }
                  }
                >
                  ,
                  90
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    bottomInset={0}
    style={
      {
        "backgroundColor": "#efefef",
        "bottom": 0,
        "paddingBottom": 12,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#1Ec677",
          "borderBottomLeftRadius": 50,
          "borderBottomRightRadius": 50,
          "borderTopLeftRadius": 50,
          "borderTopRightRadius": 50,
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontFamily": "Poppins-Bold",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Comprar
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
